### PR TITLE
Package cleanup

### PR DESCRIPTION
**Pull request by aidan-casey**
This pull request begins to clean up the package by the following:

- Updates directory structure to Laravel recommended.
- Updates version constraints to the Composer recommended caret.
- Updates testing workflows to run with lowest version of dependencies and highest stable.
- Updates configuration file with helpful comments.
- Adds tests.

### CODE REVIEW

File: config/ecotone.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    | If true, then Ecotone will cache all configurations. This will decrease
+    | application load time, but results in slower feedback for the developer
+    | as the cache will need to be cleared after changes are made. This is
+    | best suited for production.
+    |
+    | If false, then Ecotone will not cache any configurations. This will
+    | increase application load time, but results in quicker feedback for the
+    | developer. This is best suited for local development.
+    |
+    */
```
Comment (dgafka) [inline config/ecotone.php:53]: Amazing work with the description :)

**Comment by dgafka** [comment]
Thank you for putting your effort into that, once more amazing work! :)

**Comment by dgafka** [comment]
Is there a way to run whole Laravel application in test, to actually see for simple scenario that everything wires up correctly and can be executed?

There is something like this for Symfony integration. 
It uses [Behat scenario](https://github.com/ecotoneframework/symfony/blob/master/tests/Behat/features/messaging.feature) powered by [symfony-extension](friends-of-behat/symfony-extension) to boot up the Symfony Kernel (running whole Symfony App), which then can go through simple scenario.

**Comment by aidan-casey** [comment]
> Is there a way to run whole Laravel application in test, to actually see for simple scenario that everything wires up correctly and can be executed?

@dgafka - Yes. The package `orchestra/testbench` allows us to do that in PHPUnit. We can continue to flesh out those tests now that it is a dev dependency.

**Comment by dgafka** [comment]
@aidan-casey would you be open to provide PR with such test? :)

**Comment by aidan-casey** [comment]
> @aidan-casey would you be open to provide PR with such test? :)

I will look into it. I know there will be a bit more involved with that, so I just need to find the time.

**Pull request merged**